pkgs/by-name/hu/hugo/package.nix

```
@@ -69,13 +69,13 @@ buildGoModule rec {
     versionCheckHook
   ];
   doInstallCheck = true;
-  versionCheckProgram = "${placeholder "out"}/bin/${meta.mainProgram}";
+  versionCheckProgram = "${placeholder "out"}/bin/hugo";
```

```suggestion
  versionCheckProgram = "${placeholder "out"}/bin/${finalAttrs.meta.mainProgram}";
```

Why did you change this line from using mainProgram? It seems to produce the same result, and I think the old code is clearer.

---

I agree, this should be reverted.

---

I think I tricked myself into "don't `pname` on `repo`"-ing something that I shouldn't 🥴 

`/bin/hugo` is also referenced directly by the `postInstall` phase, should we change that one too?

---

The postInstall is totally fine for me... or I misunderstood your comment?

---

I mean if we should change all instances of `$out/bin/hugo` in `postInstall` to `$out/bin/${finalAttrs.meta.mainProgram}`

---

maybe related https://github.com/NixOS/nixpkgs/issues/310373

---

> I mean if we should change all instances of `$out/bin/hugo` in `postInstall` to `$out/bin/${finalAttrs.meta.mainProgram}`

No don't, keep the string hugo everywhere, meta.mainProgram or pname are not meant to be reused.

---

> No don't, keep the string hugo everywhere, meta.mainProgram or pname are not meant to be reused.

So this is solved, then? Or do we keep `meta.mainProgram` in `versionCheckProgram`?

---

Yes, sorry for the confusion, Github on mobile had a hiccup.